File: Java App/app/src/main/java/com/srivathsan/testretrofit/view/adapter/ShowsAdapter.java
package com.srivathsan.testretrofit.view.adapter;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.srivathsan.testretrofit.R;
import com.srivathsan.testretrofit.network.response.shows.Show;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class ShowsAdapter extends RecyclerView.Adapter<ShowsAdapter.MyViewHolder> {

    private List<Show> showList;

    public ShowsAdapter(List<Show> showList) {
        this.showList = showList;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.item_show, viewGroup, false);
        return new MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int i) {
        Show show = showList.get(i);
        holder.tvShow.setText(show.getShowName());
    }

    @Override
    public int getItemCount() {
        return showList.size();
    }

    public class MyViewHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.tvShow)
        TextView tvShow;

        public MyViewHolder(@NonNull View view) {
            super(view);
            ButterKnife.bind(this, view);
        }
    }
}

File: Simple Kotlin App/app/src/main/java/com/srivathsan/testretrofit/view/adapter/ShowsAdapter.kt
package com.srivathsan.testretrofit.view.adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.srivathsan.testretrofit.R
import com.srivathsan.testretrofit.network.response.shows.Show

class ShowsAdapter(private val showList: List<Show>) : RecyclerView.Adapter<ShowsAdapter.MyViewHolder>() {

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): MyViewHolder {
        val view = LayoutInflater.from(viewGroup.context)
                .inflate(R.layout.item_show, viewGroup, false)
        return MyViewHolder(view)
    }

    override fun onBindViewHolder(holder: MyViewHolder, i: Int) {
        val show = showList[i]
        holder.tvShow!!.text = show.showName
    }

    override fun getItemCount(): Int {
        return showList.size
    }

    inner class MyViewHolder(view: View) : RecyclerView.ViewHolder(view) {

        internal var tvShow: TextView? = null

        init {
            tvShow = view.findViewById(R.id.tvShow)
        }
    }
}
